net-dns/coredns/coredns-1.6.5.ebuild

```diff
@@ -0,0 +1,127 @@
+    "github.com/prometheus/common v0.7.0" # Apache-2.0
+    "github.com/prometheus/procfs v0.0.5" # Apache-2.0
+    "github.com/rcrowley/go-metrics 3113b8401b8a" # BSD-2
+    "go.etcd.io/etcd a14579fbfb1a github.com/etcd-io/etcd" # Apache-2.0
+    "github.com/coreos/go-systemd 93d5ec2c7f76" # Apache-2.0
+    "github.com/coreos/pkg 399ea9e2e55f" # Apache-2.0
+    "go.uber.org/zap v1.10.0 github.com/uber-go/zap" # MIT
+    "go.uber.org/atomic v1.3.2 github.com/uber-go/atomic" # MIT
+    "go.uber.org/multierr v1.1.0 github.com/uber-go/multierr" # MIT
+    "github.com/google/uuid v1.1.1" # BSD
+    "github.com/google/gofuzz v1.0.0" # Apache-2.0
+    "github.com/google/go-cmp v0.3.0" # BSD
+    "google.golang.org/api v0.13.0 github.com/googleapis/google-api-go-client" # BSD MIT
+    "google.golang.org/grpc v1.24.0 github.com/grpc/grpc-go" # Apache-2.0
+    "google.golang.org/genproto 710ae3a149df github.com/google/go-genproto" # Apache-2.0
+    "github.com/googleapis/gax-go v2.0.5" # BSD
+    "github.com/googleapis/gnostic v0.2.0" # Apache-2.0
+    "cloud.google.com/go v0.41.0 github.com/googleapis/google-cloud-go" # Apache-2.0
+    "go.opencensus.io v0.22.0 github.com/census-instrumentation/opencensus-go" # Apache-2.0
+    "github.com/hashicorp/golang-lru v0.5.1" # MPL-2.0
+    "gopkg.in/DataDog/dd-trace-go.v1 v1.19.0 github.com/DataDog/dd-trace-go" # BSD Apache-2.0
+    "github.com/DataDog/datadog-go v2.2.0" # MIT
+    "github.com/tinylib/msgp v1.1.0" # MIT
+    "github.com/philhofer/fwd v1.0.0" # MIT
+    "gopkg.in/inf.v0 v0.9.1 github.com/go-inf/inf" # BSD
+    "gopkg.in/square/go-jose.v2 v2.2.2 github.com/square/go-jose" # Apache-2.0
+    "gopkg.in/yaml.v2 v2.2.2 github.com/go-yaml/yaml" # Apache-2.0
+    "sigs.k8s.io/yaml v1.1.0 github.com/kubernetes-sigs/yaml" # MIT BSD
+    "k8s.io/api 7cf5895f2711 github.com/kubernetes/api" # Apache-2.0
+    "k8s.io/apimachinery 1799e75a0719 github.com/kubernetes/apimachinery" # Apache-2.0
+    "k8s.io/client-go 78d2af792bab github.com/kubernetes/client-go" # Apache-2.0
+    "k8s.io/klog v0.4.0 github.com/kubernetes/klog" # Apache-2.0
+    "k8s.io/utils 6999998975a7 github.com/kubernetes/utils" # Apache-2.0
+    "github.com/gophercloud/gophercloud fe1ba5ce12dd" # Apache-2.0
+    "github.com/imdario/mergo v0.3.7" # BSD
+    "github.com/json-iterator/go v1.1.7" # MIT
+    "github.com/modern-go/concurrent bacd9c7ef1dd" # Apache-2.0
+    "github.com/modern-go/reflect2 v1.0.1" # Apache-2.0
+    "github.com/spf13/pflag v1.0.3" # BSD
+)
+
+EGO_PN="github.com/${PN}/${PN}"
+
+inherit go-module
+
+EGIT_COMMIT="c2fd1b2467249d8b1bb6bbee72ad83b63dd9087f"
+ARCHIVE_URI="https://${EGO_PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+KEYWORDS="~amd64"
+
+DESCRIPTION="A DNS server that chains middleware"
+HOMEPAGE="https://github.com/coredns/coredns"
+
+SRC_URI="${ARCHIVE_URI}
+	$(go-module_vendor_uris)"
+LICENSE="Apache-2.0 MIT BSD ISC MPL-2.0 BSD-2"
+SLOT="0"
+IUSE=""
+
+RESTRICT="test"
```

Comment:
Can you tell me why this is restricted? I guess because it might need some network connection, but we try to leave a comment above RESTRICT to tell why it is so. You or I can add it to the ebuild.

Reply:
To be honest, I just copied this line from previous ebuild. Sure, it is network related and server is trying to start serving on 0.0.0.0:53 for example, that could be the reason.

As for licenses, I've used some tools to figure out licenses, but yes still it is a lot of manual work - for example when people integrate the license into README, or when some unrelated crap lives in the repo and that crap will have its own license(s)..

With diffs it is easier though, as I only need to inspect diffs of `go.sum` and `go.mod` and it could be even faster to check everything just using a browser.

I'm not aware of glice tool though, only what was discussed here https://bugs.gentoo.org/694792